molt/src/expr.rs

```diff
@@ -250,10 +237,10 @@ const OP_STRINGS: [&str; 36] = [
 pub fn expr(interp: &mut Interp, expr: &Value) -> MoltResult {
     let value = expr_top_level(interp, expr.as_str())?;
 
-    match value.vtype {
-        Type::Int => molt_ok!(Value::from(value.int)),
-        Type::Float => molt_ok!(Value::from(value.flt)),
-        Type::String => molt_ok!(Value::from(value.str)),
+    match value {
+        Datum::Int(v) => molt_ok!(Value::from(v)),
+        Datum::Float(v) => molt_ok!(Value::from(v)),
+        Datum::String(v) => molt_ok!(Value::from(v)),
     }
```

Comment:
I'm not sure if this would work, but it might be worth trying
```suggestion
    molt_ok!(match value {
        Datum::Int(v) => Value::from(v),
        Datum::Float(v) => Value::from(v),
        Datum::String(v) => Value::from(v),
    })
```